test/language/expressions/async-function/try-throw-finally-return.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 Caitlin Potter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Caitlin Potter <caitp@igalia.com>
+esid: pending
+description: >
+  Implementations must defer rejecting an async function's Promise until after
+  all finally blocks have been evaluated.
+flags: [async]
+---*/
+
+var f = async function() {
+  try {
+    throw "early-throw";
+  } finally {
+    return await new Promise(function(resolve, reject) {
+      resolve("override");
+    });
+  }
+};
+
+f().then($DONE, function(value) {
+  assert.sameValue("override", value);
```

Comment:
As above, this callback should be specified as the first argument to `then`, not the second.